core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,131 @@
+    }
+  }
+
+  public get arc(): Arc3d {
+    return this._arc;
+  }
+  public get vector0LengthSquared(): number {
+    return this._axx;
+  }
+  public get vector90LengthSquared(): number {
+    return this._ayy;
+  }
+  public get curvatureRange(): Range1d {
+    return this._curvatureRange;
+  }
+  public get isValidArc(): boolean {
+    return this._isValidArc;
+  }
+
+  /**
+   * Compute the angle corresponding to the point in the ellipse's first quadrant with the given curvature.
+   * * The instance sweep is ignored.
+   * * Arc is assumed to be non-circular and have perpendicular axes of positive length.
+   * * This is a scaled inverse of [[Arc3d.fractionToCurvature]] restricted to fractions in [0,1/4].
+   * @return radian angle in [0,pi/2] or undefined if the ellipse is invalid, or does not attain the given curvature.
+  */
+  private curvatureToAngle(curvature: number): number | undefined {
+    if (!this.isValidArc)
+      return undefined;
+    if (!this.curvatureRange.containsX(curvature))
+      return undefined; // ellipse does not attain this curvature
+    const scaledNormalLengthSquared = this._axx * this._ayy / curvature * curvature;
+    const numerator = Math.cbrt(scaledNormalLengthSquared) - this._axx;
+    const denominator = this._ayy - this._axx;
+    const cosTheta = Math.sqrt(Math.abs(numerator / denominator));
+    return Math.acos(cosTheta);
+  }
+
+  /**
+   * Sample the instance arc by interpolating curvature, and return the fractions of the sample points.
+   * @param numPointsInQuadrant number of samples in each ellipse quadrant.
+   * @param structuredOutput false to return a flat array of fractions (default). If true, structured objects are returned:
+   * * Fractions are sorted and assembled by ellipse quadrant into a [[QuadrantFractions]] object.
+   * * If the arc sweep spans adjacent quadrants, `QuadrantFractions`representing those quadrants share the fraction at the quadrant boundary.
+   * * `QuadrantFractions` objects are sorted in increasing fraction order.
+   * * Note that if the arc starts and ends in the same ellipse quadrant, two `QuadrantFractions` objects can be returned.
+   */
+  private sampleFractions(numPointsInQuadrant: number = 4, _structuredOutput: boolean = false): QuadrantFractions[] | number[] {
+    if (numPointsInQuadrant < 2)
+      return [];
+    const result: QuadrantFractions[] = [];
+
+/*    const f = (x: number): number => { return x; }; // TODO
+
+  // don't add interior fraction if near quadrant fraction
+
+    const cMin = this.curvatureRange.low;
+    const cMax = this.curvatureRange.high;
+    const tDelta = 1.0 / (numPointsInQuadrant - 1);
+    for (let i = 1; i < numPointsInQuadrant; ++i) {
+      const j = f(i * tDelta);
+      const curvature = (1 - j) * cMin + j * cMax;
+      for (const fraction of curvatureToFractions(ellipticalArc, curvature))
+        fractions.add(fraction);
+    }
+    for (const fraction of [...fractions].sort())
```

Comment:
As discussed, we should also drop fractions that are too close together because they are unnecessary and reduce the performance. I was able to do that by doing below rather than a simple `[...fractions].sort()`

``` 
    let tolerance = 0.01;
    const filteredFractions = [...fractions].sort((a, b) => a - b).filter((val, i, arr) => {
      return i === 0 || Math.abs(val - arr[i - 1]) >= tolerance;
    });
```

You may want to chaneg the tolerance but I think 0.01 (which is 1%) is good.

Reply:
Near duplicates should be disallowed by the comparison function for the relevant `OrderedSet` objects.